lib/rpmts.c

```diff
@@ -561,9 +566,17 @@ rpmRC rpmtsImportPubkey(const rpmts ts, const unsigned char * pkt, size_t pktlen
     int subkeysCount = 0;
     rpmVSFlags oflags = rpmtsVSFlags(ts);
     rpmKeyring keyring;
-    rpmtxn txn = rpmtxnBegin(ts, RPMTXN_WRITE);
     int krc, i;
 
+    char *krtype = rpmExpand("%{?_keyring}", NULL);
+    if (rstreq(krtype, "fs")) {
```

Comment:
Since we now need this in two separate places, add an internal helper to retrieve the keyring type as an enum or such  so callers don't need to bother with the details of macro expansion and freeing the results. 

The other question is whether it wouldn't be just as simple to actually support import rather than deny it, as surprisingly often is the case :sweat_smile: Did you look at that option at all? Not saying it's a must, just curious.

Reply:
I'll follow up in #1962